aiida_common_workflows/workflows/relax/vasp/generator.py

```diff
@@ -0,0 +1,145 @@
+# -*- coding: utf-8 -*-
+"""Implementation of `aiida_common_workflows.common.relax.generator.RelaxInputGenerator` for VASP."""
+from aiida import orm
+
+from ..generator import RelaxInputsGenerator, RelaxType
+from .workchain import CommonVASPRelaxWorkChain
+
+__all__ = ('VASPRelaxInputsGenerator',)
+
+
+class VASPRelaxInputsGenerator(RelaxInputsGenerator):
+    """Input generator for the `VASPRelaxWorkChain`."""
+
+    _default_protocol = 'efficiency'
+    _protocols = {'efficiency': {'description': ''}, 'precision': {'description': ''}}
+
+    _calc_types = {'relax': {'code_plugin': 'vasp.vasp', 'description': 'The code to perform the relaxation.'}}
+
+    _relax_types = {
+        RelaxType.ATOMS: 'Relax only the atomic positions while keeping the cell fixed.',
+        RelaxType.ATOMS_CELL: 'Relax both atomic positions and the cell.'
+    }
+
+    def get_builder(
+        self,
+        structure,
+        calc_engines,
+        protocol,
+        relaxation_type,
+        threshold_forces=None,
+        threshold_stress=None,
+        **kwargs
+    ):
+        """Return a process builder for the corresponding workchain class with inputs set according to the protocol.
+
+        :param structure: the structure to be relaxed
+        :param calc_engines: ...
+        :param protocol: the protocol to use when determining the workchain inputs
+        :param relaxation_type: the type of relaxation to perform, instance of `RelaxType`
+        :param threshold_forces: target threshold for the forces in eV/Å.
+        :param threshold_stress: target threshold for the stress in eV/Å^3.
+        :param kwargs: any inputs that are specific to the plugin.
+        :return: a `aiida.engine.processes.ProcessBuilder` instance ready to be submitted.
+        """
+        # pylint: disable=too-many-locals
+        code = calc_engines['relax']['code']
+        options = calc_engines['options']
+        settings = calc_engines['settings']
+
+        try:
+            potential_family = settings['potential_family']
+        except KeyError:
+            potential_family = 'PBE'
+
+        try:
+            potential_mapping = settings['potential_mapping']
+        except KeyError:
+            potential_mapping = 'DEFAULT_PBE'
+
+        process_class = CommonVASPRelaxWorkChain._process_class  # pylint: disable=protected-access
+
+        builder = CommonVASPRelaxWorkChain.get_builder()
+
+        # Set structure
+        builder.structure = structure
+
+        # Set options
+        inputs.options = DataFactory('dict')(dict=options)
+
+        # Set settings
+        # Make sure we add forces and stress for the parser
+        settings.update({'parser_settings': {'add_forces': True, 'add_stress': True}})
+        inputs.settings = DataFactory('dict')(dict=settings)
```

Comment:
I hope I coded the `DataFactory` to be clever enough an use a cache, but you never know ;) maybe it would be better to load these classes once and just reuse the class. Also, for the built-in types into `aiida-core` (`Dict`, `Float`, `Bool`, `Str`, `Int` etc) I think using direct import from `aiida.orm` is fine. Note that for `StructureData` I *would* use the factory, since that one is domain specific and will be moved at some point

Reply:
Yes, I hope so too :) Comments noted.